control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2071,6 +2077,36 @@ func (r *HostedControlPlaneReconciler) reconcileClusterNetworkOperator(ctx conte
 	return nil
 }
 
+// reconcileDNSOperator ensures that the management cluster has the expected DNS
+// operator deployment and kubeconfig secret for the hosted cluster.
+func (r *HostedControlPlaneReconciler) reconcileDNSOperator(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage, createOrUpdate upsert.CreateOrUpdateFN) error {
+	p := dnsoperator.NewParams(hcp, releaseImage.Version(), releaseImage.ComponentImages())
+
+	rootCASecret := manifests.RootCASecret(hcp.Namespace)
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(rootCASecret), rootCASecret); err != nil {
+		return err
+	}
+
+	kubeconfig := manifests.DNSOperatorKubeconfig(hcp.Namespace)
+	if _, err := createOrUpdate(ctx, r, kubeconfig, func() error {
+		return pki.ReconcileServiceAccountKubeconfig(kubeconfig, rootCASecret, hcp, "openshift-dns-operator", "dns-operator")
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile dnsoperator kubeconfig: %w", err)
+	}
+
+	deployment := manifests.DNSOperatorDeployment(hcp.Namespace)
+	if _, err := createOrUpdate(ctx, r, deployment, func() error {
+		// XXX Should we log or return errors, or remove the return
+		// value from ReconcileDeployment?
+		dnsoperator.ReconcileDeployment(deployment, p, util.APIPort(hcp))
```

Comment:
if `ReconcileDeployment` returns an error it should be returned here, but from what I can see while it has an error return that return is always nil? Maybe just remove it?

Reply:
I'll remove the return value.